### PR TITLE
Decompile DRA EntityStopWatch

### PR DESCRIPTION
**Pull request by joshlory**
PSX: https://decomp.me/scratch/eMQyN
PSP: https://decomp.me/scratch/1Ktyt

### CODE REVIEW

**Comment by joshlory** [comment]
Need to fix up the rodata offsets.

File: src/dra/843B0.c [hunk 1]
```diff
@@ -2498,7 +2498,355 @@ void EntityStopWatchExpandingCircle(Entity* self) {
...
+        }
+    }
+
+    switch (self->step) {
+    case 0:
+        if (g_unkGraphicsStruct.unk0) {
+            DestroyEntity(self);
+            return;
+        }
+        self->primIndex = AllocPrimitives(4, 3);
```
Comment (Xeeynamo) [inline src/dra/843B0.c:2524]:
```suggestion
        self->primIndex = AllocPrimitives(PRIM_GT4, 3);
```

File: src/dra/843B0.c [hunk 2]
```diff
@@ -2498,7 +2498,355 @@ void EntityStopWatchExpandingCircle(Entity* self) {
...
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        self->flags = 0x0C860000;
```
Comment (Xeeynamo) [inline src/dra/843B0.c:2529]: Could you document this one with the flags from `game.h` at line `253`? For example `0x00800000` is `FLAG_HAS_PRIMS`. There are other places like `prim->drawMode` too

**Review by Xeeynamo — COMMENTED** [review]
Left a couple of comments related to documenting some magic numbers. I cannot run the CI until the conflicts get solved. For the rodata you will need to import the jtbl on both US and HD of DRA

**Comment by joshlory** [comment]
The rodata offset ended up being an indication of a file break (thanks @bismurphy!). I _think_ this one should be ready to review.

**Pull request merged**